src/lib/Context.ts

```
@@ -119,14 +147,19 @@ export function consolidateConfigs<T>(...configs: Partial<ConfigJson>[]): Config
-  const signer = await createSignerFromPath(config.keypair!)
+  // if transactionContext is true, we need to make sure we have a valid keypair either in the config or overrides
+  if (transactionContext === true && !config.keypair && !overrides.keypair) {
+    throw new Error('No keypair specified in config or args. Please set a keypair path in your config file or use --keypair flag.')
+  }
+
+  const signer = config.keypair ? await createSignerFromPath(config.keypair) : createNoopSigner()
 
   const payer = config.payer ? await createSignerFromPath(config.payer) : signer
 
-  const umi = createUmi(config.rpcUrl!, {
+  const umi = createUmi(config.rpcUrl! , {
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)